source/Octopus.Tentacle.Tests.Integration/TentacleCommandLineTests.cs

```diff
@@ -0,0 +1,659 @@
+                Logger.Error($"Command StdOut: {Environment.NewLine}{stdout}");
+                Logger.Error($"Command StdErr: {Environment.NewLine}{stderr}");
+
+                throw;
+            }
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task HelpAsFirstArgumentShouldShowCommandSpecificHelp(TentacleConfigurationTestCase tc)
+        {
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, "help version");
+            stderr.Should().BeNullOrEmpty();
+
+            stdout.Should().Be(
+@"Usage: Tentacle version [<options>]
+
+Where [<options>] is any of: 
+
+      --format=VALUE         The format of the output (text,json). Defaults 
+                               to text.
+
+Or one of the common options: 
+
+      --help                 Show detailed help for this command
+");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task ShowConfigurationCommand(TentacleConfigurationTestCase tc)
+        {
+            await using var clientAndTentacle = await tc.CreateBuilder().Build(CancellationToken);
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, $"show-configuration --instance={clientAndTentacle.RunningTentacle.InstanceName}");
+            stderr.Should().BeNullOrEmpty();
+
+            // Actually parse and query the document just like our consumer will
+            dynamic? settings = JsonConvert.DeserializeObject(stdout);
+
+            ((string)settings.Octopus.Home).Should().Be(clientAndTentacle.RunningTentacle.HomeDirectory, "the home directory should match");
+            ((string)settings.Tentacle.Deployment.ApplicationDirectory).Should().Be(clientAndTentacle.RunningTentacle.ApplicationDirectory, "the application directory should match");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task ShowConfigurationCommandOnPartiallyConfiguredTentacle(TentacleConfigurationTestCase tc)
+        {
+            var instanceId = Guid.NewGuid().ToString();
+            using var temporaryDirectory = new TemporaryDirectory();
+            await RunCommandWithValidExitCodeAssert(tc, $"create-instance --instance={instanceId} --config=\"{Path.Combine(temporaryDirectory.DirectoryPath, instanceId + ".cfg")}\"");
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, $"show-configuration --instance={instanceId}");
+            stderr.Should().BeNullOrEmpty();
+
+            // Actually parse and query the document just like our consumer will
+            dynamic? settings = JsonConvert.DeserializeObject(stdout);
+            ((string)settings.Octopus.Home).Should().Be(temporaryDirectory.DirectoryPath, "the home directory should match");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        [NonParallelizable]
```

Comment:
Why is this set? Not that it is an issue to set it.

Reply:
Some of the tests collide and cause others to fail e.g. the watchdog tests change the json config file output. Isolated the tests that cause issues so they run in isolation.